### ISSUE TITLE
Explorer Crashing on 22000.258

### ISSUE THREAD

**Issue by washcroft** [valinet/ExplorerPatcher#85]
Hello,
Just updated Windows to 22000.258 and ExplorerPatcher to 22000.258.26.1 - however explorer is now crashing when the DLL is in place. I have tried resetting settings in the UI to defaults but still happens. I had the notification to say symbols downloaded and install complete on first run.

```
Faulting application name: explorer.exe, version: 10.0.22000.120, time stamp: 0xe846e749
Faulting module name: ucrtbase.dll, version: 10.0.22000.1, time stamp: 0x00e78ce9
Exception code: 0xc0000409
Fault offset: 0x000000000007dd7e

Fault bucket 1983653253604524220, type 5
Event Name: BEX64
Response: Not available
Cab Id: 0

Problem signature:
P1: explorer.exe
P2: 10.0.22000.120
P3: e846e749
P4: ucrtbase.dll
P5: 10.0.22000.1
P6: 00e78ce9
P7: 000000000007dd7e
P8: c0000409
P9: 0000000000000007
P10: 
```

**Comment by valinet** [comment]
I understand, if no one want to attach a debugger, screenshots and that summary is not useful in anyway. All I can tell is the error is somewhere in the Microsoft Visual C runtime library. I personally cannot reproduce the error. Even if one wants to give me TeamViewer access, I can login and debug it on your machine, it simply does not happen on my side, no matter how I try to reproduce it.

**Comment by valinet** [comment]
> 
> 
> > eventually it tries to restart explorer.exe automatically or manually sometimes if I run `rundll32.exe C:\Windows\dxgi.dll,ZZGUI`. Once the explorer closes it can't be opened again until the .dll is deleted and the computer restarted so that the Win11 taskbar is used again
> 
> Same here, it is intermittent and will sometimes work, but restarting explorer through the dxgi UI will often cause the issue.
> 
> WinDbg analysis attached
> 
> [explorer.exe.30188.dmp.txt](https://github.com/valinet/ExplorerPatcher/files/7339646/explorer.exe.30188.dmp.txt)

Thanks.

Can we please establish a clear pattern of what triggers this? Let's forget about restarting via the GUI for a second, does it work when Explorer simply launches at logon? Is it stable there, or does it crash after a few minutes? Does it sometimes work or it crashes there at all times? Or does this only happen when using the "restart Explorer" option in the GUI?

Idk, from the stack trace, it's in some function that determines some theme for the taskbar... I never mess with that, but anyway... themes being the key work, do you use light/dark theme for system/apps, do you have a uxtheme patcher installed etc? Idk where to start on this, it seems to be some random problem in Explorer...

Can you try with the attached DLL as well?

[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7339726/dxgi.zip)

**Comment by valinet** [comment]
> 
> 
> > I attach a DLL which changes the behavior of "Restart File Explorer" in the GUI to just exit Explorer and let the system restart it, can you test with that as well, please? [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7339817/dxgi.zip)
> 
> That one doesn't seem to restart explorer, it just closes the UI dialog.

Depends on where you run it from. Do you run the GUI in `rundll32.exe` or from right click Start - Properties. That is important as well.

**Comment by washcroft** [comment]
> > Can you try with the attached DLL as well?
> > [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7339726/dxgi.zip)
> 
> For the first time, I cannot reproduce the issue using this.

This one still working fine by the way... explorer restarts fine despite the GUI being open.

**Comment by washcroft** [comment]
> Edit: My AV is also blocking the github links and preventing me from downloading the .dll right now.

Interesting, I deleted my post thinking I'd just confused it with all the DLL testing switch-a-roo.

**Comment by washcroft** [comment]
> Can you disable the antivirus altogether and test, please?

Makes no difference, still crashing with Defender disabled.

**Comment by valinet** [comment]
So, does this still crash: 
[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7340102/dxgi.zip)
?

**Comment by valinet** [comment]
[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7340148/dxgi.zip)

**Comment by valinet** [comment]
Aha, okay, now I understand. Idk, I did not get it from the first read, it made sense but my brain just stopped. Anyway, so, do you think we can consider this solved now? The fix is a mix of fixing for #78 while also making sure the problem here does not happen. I hope it still works well for #78, probably it does. Do you think this can be marked as fixed, does it happen again at all, say, after a restart, or whatever you think might make it crack again?

**Comment by valinet** [comment]
This was a very nasty one. I almost broke a monitor =)))) the keyboard is long gone... Thanks for the support throughout and for the prompt replies @washcroft and @Kyrex; even though I get nervous some times, it comes from the original desire to help, so thanks for bearing with me.
